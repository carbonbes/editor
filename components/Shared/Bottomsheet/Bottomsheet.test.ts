import { describe, expect, test } from 'vitest'
import { flushPromises, mount } from '@vue/test-utils'
import {
  BottomsheetContent,
  BottomsheetContentHeader,
  BottomsheetContentScrollable,
  BottomsheetPortal,
  BottomsheetRoot,
} from './'

async function mountBottomsheet(template?: string) {
  const Component = defineComponent({
    components: {
      BottomsheetContent,
      BottomsheetContentHeader,
      BottomsheetContentScrollable,
      BottomsheetPortal,
      BottomsheetRoot,
    },
    template: `
      <BottomsheetRoot v-model:open="open">
        <BottomsheetPortal>
          ${template}
        </BottomsheetPortal>
      </BottomsheetRoot>
    `,
    setup() {
      const open = ref(false)

      return { open }
    },
  })

  const wrapper = mount(Component, { props: { open: true } })

  await flushPromises()

  return wrapper
}

describe('Bottomsheet', () => {
  describe('BottomsheetContent', () => {
    const cases = [
      ['should render without slot', '<BottomsheetContent />'],
      [
        'should render with slot',
        '<BottomsheetContent><p>Bottomsheet Content Test</p></BottomsheetContent>',
      ],
    ]

    test.each(cases)('%s', async (_, template) => {
      const wrapper = await mountBottomsheet(template)
      const body = wrapper.getComponent(BottomsheetContent)

      expect(body.html()).toMatchSnapshot()

      wrapper.unmount()
    })
  })

  describe('BottomsheetContentHeader', () => {
    const cases = [
      ['should render without slot', '<BottomsheetContentHeader />'],
      [
        'should render with slot',
        '<BottomsheetContentHeader><p>Bottomsheet Content Header Test</p></BottomsheetContentHeader>',
      ],
    ]

    test.each(cases)('%s', async (_, template) => {
      const wrapper = await mountBottomsheet(template)
      const header = wrapper.getComponent(BottomsheetContentHeader)

      expect(header.html()).toMatchSnapshot()

      wrapper.unmount()
    })

    test('must have a close button', async () => {
      const wrapper = await mountBottomsheet('<BottomsheetContentHeader />')
      const header = wrapper.getComponent(BottomsheetContentHeader)
      const button = header.get(
        '[data-testid="bottomsheet-content-header/close-button"]',
      )

      expect(button).toBeDefined()

      wrapper.unmount()
    })

    test('clicking on the close button should close the Bottomsheet', async () => {
      const wrapper = await mountBottomsheet('<BottomsheetContentHeader />')
      const header = wrapper.getComponent(BottomsheetContentHeader)
      const button = header.get(
        '[data-testid="bottomsheet-content-header/close-button"]',
      )

      expect(wrapper.vm.open).toBe(true)

      await button.trigger('click')

      expect(wrapper.vm.open).toBe(false)

      wrapper.unmount()
    })
  })

  describe('BottomsheetContentScrollable', () => {
    const cases = [
      ['should render without slot', '<BottomsheetContentScrollable />'],
      [
        'should render with slot',
        '<BottomsheetContentScrollable><p>Test</p></BottomsheetContentScrollable>',
      ],
    ]

    test.each(cases)('%s', async (_, template) => {
      const wrapper = await mountBottomsheet(template)
      const content = wrapper.getComponent(BottomsheetContentScrollable)

      expect(content.html()).toMatchSnapshot()

      wrapper.unmount()
    })
  })
})
