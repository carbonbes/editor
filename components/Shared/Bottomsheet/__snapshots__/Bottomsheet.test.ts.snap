// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bottomsheet > BottomsheetContent > should render with slot 1`] = `
"<div data-dismissable-layer="" style="--snap-point-height: 0; pointer-events: auto;" id="" role="dialog" aria-describedby="" aria-labelledby="reka-dialog-title-v-0" data-state="open" vaul-drawer="" vaul-drawer-direction="bottom" vaul-drawer-visible="true" tabindex="-1" class="fixed right-0 bottom-0 left-0 w-full h-full max-h-[95%] flex flex-col bg-white rounded-t-3xl"><span aria-hidden="true" style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0, 0, 0, 0); clip-path: inset(50%); white-space: nowrap; word-wrap: normal;"><h2 id="reka-dialog-title-v-0"></h2></span>
  <p>Bottomsheet Content Test</p>
</div>"
`;

exports[`Bottomsheet > BottomsheetContent > should render without slot 1`] = `"<div data-dismissable-layer="" style="--snap-point-height: 0; pointer-events: auto;" id="" role="dialog" aria-describedby="" aria-labelledby="reka-dialog-title-v-0" data-state="open" vaul-drawer="" vaul-drawer-direction="bottom" vaul-drawer-visible="false" tabindex="-1" class="fixed right-0 bottom-0 left-0 w-full h-full max-h-[95%] flex flex-col bg-white rounded-t-3xl"><span aria-hidden="true" style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0, 0, 0, 0); clip-path: inset(50%); white-space: nowrap; word-wrap: normal;"><h2 id="reka-dialog-title-v-0"></h2></span></div>"`;

exports[`Bottomsheet > BottomsheetContentHeader > should render with slot 1`] = `
"<div class="flex items-center">
  <p>Bottomsheet Content Header Test</p><button type="button" class="ml-auto p-1.5 flex items-center justify-center bg-gray-200/50 rounded-full" data-testid="bottomsheet-content-header/close-button"><span class="iconify i-tabler:x size-5" aria-hidden="true"></span></button>
</div>"
`;

exports[`Bottomsheet > BottomsheetContentHeader > should render without slot 1`] = `"<div class="flex items-center"><button type="button" class="ml-auto p-1.5 flex items-center justify-center bg-gray-200/50 rounded-full" data-testid="bottomsheet-content-header/close-button"><span class="iconify i-tabler:x size-5" aria-hidden="true"></span></button></div>"`;

exports[`Bottomsheet > BottomsheetContentScrollable > should render with slot 1`] = `
"<div class="overflow-y-auto">
  <p>Test</p>
</div>"
`;

exports[`Bottomsheet > BottomsheetContentScrollable > should render without slot 1`] = `"<div class="overflow-y-auto"></div>"`;
